setup.py

```diff
@@ -47,6 +47,8 @@ def build_extension(self, ext):
             '-DPYTHON_EXECUTABLE=' + sys.executable,
             '-DVARA_FEATURE_USE_Z3_SOLVER=True'
         ]
+        if os.path.exists('/lib/x86_64-linux-gnu/libz3.so'):
+            cmake_args.append('-DVARA_FEATURE_BUILD_Z3_SOLVER=False')
```

Comment:
I'm not sure if this is a good way to determine if we should build the solver.
1) the path is a system specific
2) we might want to build the solver from scratch even for those system, as we depend on a very specific version of z3. (The debian 11 can cause problems)

Maybe we should keep building the solver as the default and give the user a option to disable that? Or, if it works for you, only you use the local one.